### PR TITLE
ResourceLeakDetector

### PR DESCRIPTION
**Pull request by StormHub**
1. Leak records target amount and backing off
2. Composite byte buffer copy.
3. Fixed false positive resource leak report sometimes #308 
4. Fixed ByteBufferUtil for string read
5. Removed unused EmptyByteBuffer methods

### CODE REVIEW

File: src/DotNetty.Buffers/ByteBufferUtil.cs [hunk 1]
```diff
@@ -360,7 +360,7 @@ public static string DecodeString(IByteBuffer src, int readerIndex, int len, Enc
                 try
                 {
                     buffer.WriteBytes(src, readerIndex, len);
-                    ArraySegment<byte> ioBuf = src.GetIoBuffer();
+                    ArraySegment<byte> ioBuf = buffer.GetIoBuffer();
```
Comment (nayato) [inline src/DotNetty.Buffers/ByteBufferUtil.cs]: I assume it means using composite buffers and ToString is broken in 0.4.7, right?

File: src/DotNetty.Common/ResourceLeakDetector.cs [hunk 2]
```diff
@@ -14,17 +14,16 @@ namespace DotNetty.Common
     using DotNetty.Common.Internal;
     using DotNetty.Common.Internal.Logging;
     using DotNetty.Common.Utilities;
-    using Nito;
 
-    public sealed class ResourceLeakDetector
```
Comment (nayato) [inline src/DotNetty.Common/ResourceLeakDetector.cs]: why not seal it?

File: src/DotNetty.Buffers/ByteBufferUtil.cs (hunk 1 above)
Comment (StormHub) [reply src/DotNetty.Buffers/ByteBufferUtil.cs]: Yes, found it in http branch, should be buffer to copy from.

File: src/DotNetty.Common/ResourceLeakDetector.cs (hunk 2 above)
Comment (StormHub) [reply src/DotNetty.Common/ResourceLeakDetector.cs]: It is not final in netty but nothing extends it for now as far as I know of.

**Pull request merged**